### ISSUE TITLE
jabber.py: Message Encription via Omemo (XEP-0384)

### ISSUE THREAD

**Issue by 0xErnie** [weechat/scripts#187]
With [OMEMO](https://xmpp.org/extensions/xep-0384.html) having conversations with multiple clients is easy and secure. It supersedes OTR.

You can check the progress of other xmpp implementations at https://omemo.top/

**Comment by wnereiz** [comment]
If there's OMEMO support I will definitely abandon bitlbee and OTR, and use the native solution (weechat+jabber.py). I used to plan to work on libotr support for weechat but I decided to switch to OMEMO after I red some papers about it.

**Comment by physkets** [comment]
@wnereiz Since there has already been a lot of work on the [purple plugin for OMEMO](https://github.com/manchito/libpurple-omemo-plugin), and because bitlbee can use libpurple, they do seem to be attempting to integrate that.

**Comment by wnereiz** [comment]
@physkets For me the problem is that I feel bitlbee is not very easy to use with weechat together - especially the intigration. For example it's command format is completely different format weechat, and really hard to remember. It is painful to get used to it.

**Comment by physkets** [comment]
I don't understand. One needs Bitlbee commands only for the first time one sets it up. After that, one may use regular Weechat commands. That is what makes it very simple to use.

After setting it up, I just use /connect to open and connect to the bitlbee buffers. You can then use /join to go to rooms and /msg to  talk to specific people. All these are Weechat commands.

**Comment by wnereiz** [comment]
@physkets Yes, but I add different buddies and join different xmpp rooms (you have to use chat add command for that) quite often. But anyway I know it is not important. You are right, once OMEMO support for libpurple will be done, I can use it with bitlbee.

**Comment by physkets** [comment]
@wnereiz I have rooms (groups) set so that it automatically `/join`s them when I `/connect` to bitlbee. If you already have the room `add`ed, just set weechat to auto-connect to it, just like an irc channel.

**Comment by weechatter** [comment]
No. for jabber its recommended to use bitlbee

**Comment by gibix** [comment]
ok, thanks

**Comment by nolash** [comment]
@weechatter It seems that the omemo dev for libpurple is stalling. The two projects I could find haven't had activity for 8 months or more.

* https://github.com/gkdr/lurch
* https://github.com/manchito/libpurple-omemo-plugin

Are you aware of any other initiatives. And if not, is still the conclusion that this is deferred to someone else?

**Comment by codeurimpulsif** [comment]
Hi, any work in progress on this issue ?
I don't understand responses with Bitlbee (is it the IRC client you're speaking about ?)

**Comment by flashcode** [comment]
@codeurimpulsif : nothing new for this feature, I still recommend to use bitlbee, this jabber.py script has some problems (like blocking WeeChat on network operations).
Bitlbee is not an IRC client, it's a gateway between many instant protocols and IRC (so you can consider it as an IRC server), see: https://www.bitlbee.org/

**Comment by ctrlcctrlv** [comment]
No progress is being made towards OMEMO in Bitlbee; @dequis over on their tracker in issue [№1239](https://bugs.bitlbee.org/ticket/1239) basically said he has no interest unless paid.* If I fix the issues, would you mind merging this @flashcode? I have one script here already but I wrote it a long long time ago: [noirccolors.py](https://github.com/weechat/scripts/blob/8f017d7bd4e4539795d6338511e5a645bce42816/python/noirccolors.py#L4). I am sure I could write something that would be performant.

* In re Bitlbee, I am trying to _avoid_ diving into foreign (i.e. not my own) C codebases as they are such timesinks, but @dequis did write:

> Unless we have a contract for an appropriate hourly rate, don't act like you're entitled to my time. 

I might be able to offer some kind of grant for the work but I really don't know technically why we are just assuming Bitlbee is the better solution. It's true that Bitlbee exploits IRC protoccol to be more instantly usable on many clients, but because of this, it can't relay a lot of the features of the underlying protocols with any fidelity. Here's an example: images. Kitty (`xterm-kitty`) can support them easily. But Bitlbee-IRC will never be able to support them for e.g. Twitter DM or FB Messenger as anything but links.

Perhaps, @flashcode, what we _actually_ want is a modern Bitlbee that translates other protocols to _XMPP_ for private consumption, rather than IRC. Does such a thing exist?

**Comment by dequis** [comment]
> @dequis over on their tracker in issue [№1239](https://bugs.bitlbee.org/ticket/1239) basically said he

*she

> Perhaps, @flashcode, what we _actually_ want is a modern Bitlbee that translates other protocols to _XMPP_ for private consumption, rather than IRC. Does such a thing exist?

https://github.com/SpectrumIM/spectrum2

(but i don't see how this makes anything easier? unless your plan is to switch from weechat to profanity)

**Comment by ctrlcctrlv** [comment]
@dequis 

> *she

Sorry !

> (but i don't see how this makes anything easier? unless your plan is to switch from weechat to profanity)

Not at all. But, well, then, either XMPP support could be added to WeeChat core or via a script if it's not done already, or Bitlbee could support it for the time being via its own XMPP support.
